http-api/src/lib.rs

```diff
@@ -475,16 +505,43 @@ async fn history_handler(
             })
             .skip(page * per_page)
             .take(per_page)
-            .cloned()
+            .map(|header| {
+                let id = commit_hash_to_fingerprint(ctx.clone(), &header.sha1.to_string())
+                    .unwrap_or_default();
```

Comment:
This will return `""` on failure, which we don't want. I'd return an error a 500 from the history handler in this case, because it means something is wrong with our git installation or our code.